scarb/src/ops/fmt.rs

```diff
@@ -0,0 +1,185 @@
+use anyhow::Result;
+use cairo_lang_formatter::{CairoFormatter, FormatOutcome, FormatterConfig};
+use ignore::WalkState::Continue;
+use ignore::{DirEntry, Error, ParallelVisitor, ParallelVisitorBuilder, WalkState};
+use std::fmt::Display;
+use std::path::Path;
+use std::sync::atomic::{AtomicBool, Ordering};
+use tracing::{info, warn};
+
+use crate::core::workspace::Workspace;
+use crate::core::{Package, PackageName};
+
+#[derive(Debug)]
+pub struct FmtOptions {
+    pub check: bool,
+    pub pkg_name: Option<PackageName>,
+    pub color: bool,
+}
+
+#[tracing::instrument(skip_all, level = "debug")]
+pub fn format(opts: FmtOptions, ws: &Workspace<'_>) -> Result<bool> {
+    console::set_colors_enabled(opts.color);
+    if let Some(pkg_name) = opts.pkg_name.clone() {
+        // Format single package by name.
+        format_package_by_cond(ws, &opts, &|pkg: &Package| pkg.id.name == pkg_name)
+    } else {
+        // Format project members.
+        format_package_by_cond(ws, &opts, &|_pkg: &Package| true)
+    }
+}
+
+struct PathFormatter<'t> {
+    all_correct: &'t AtomicBool,
+    opts: &'t FmtOptions,
+    fmt: &'t CairoFormatter,
+    ws: &'t Workspace<'t>,
+}
+
+struct PathFormatterBuilder<'t> {
+    all_correct: &'t AtomicBool,
+    opts: &'t FmtOptions,
+    fmt: &'t CairoFormatter,
+    ws: &'t Workspace<'t>,
+}
+
+impl<'s, 't> ParallelVisitorBuilder<'s> for PathFormatterBuilder<'t>
+where
+    't: 's,
+{
+    fn build(&mut self) -> Box<dyn ParallelVisitor + 's> {
+        Box::new(PathFormatter {
+            all_correct: self.all_correct,
+            opts: self.opts,
+            fmt: self.fmt,
+            ws: self.ws,
+        })
+    }
+}
+
+fn print_diff(ws: &Workspace<'_>, path: &Path, diff: impl Display) {
+    ws.config()
+        .ui()
+        .print(format!("Diff in file {}:\n {}", path.display(), diff));
+}
+
+fn print_error(ws: &Workspace<'_>, path: &Path, error: anyhow::Error) {
+    let error_msg = error.to_string();
+    ws.config().ui().error(format!(
+        "{}Error writing files: cannot parse {}",
+        // TODO(maciektr): Fix this with proper upstream changes.
+        // The slice is a hacky way of avoiding duplicated "error: " prefix.
```

Comment:
```suggestion
        // TODO(maciektr): Fix this with proper upstream changes.
        //   The slice is a hacky way of avoiding duplicated "error: " prefix.
```

IntelliJ can handle multiline TODO comments provided lines are indented :)